subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -506,7 +475,7 @@ public Collection<String> getBuildTasks() {
 
         @Override
         public FileCollection getRuntimeClasspath() {
-            SourceSet mainSourceSet = mainSourceSetOf(extension);
+            SourceSet mainSourceSet = extension.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
```

Comment:
Could pass the `mainSourceSet` down to this method via `configureBuild(Project)` and avoid re-querying it here as well.